### PR TITLE
Set identity related env vars in create dev cluster

### PR DESCRIPTION
**Pull request by nader-ziada**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
- set env variables needed for AzureClusterIdentity and secret in create-dev-cluster

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
none
```

### CODE REVIEW

File: hack/create-dev-cluster.sh [hunk 1]
```diff
@@ -50,6 +50,11 @@ export WORKER_MACHINE_COUNT=${WORKER_MACHINE_COUNT:-2}
 export KUBERNETES_VERSION="${KUBERNETES_VERSION:-v1.19.7}"
 export CLUSTER_TEMPLATE="${CLUSTER_TEMPLATE:-cluster-template.yaml}"
 
+# identity secret settings.
+export AZURE_CLUSTER_IDENTITY_SECRET_NAME="cluster-identity-secret"
```
Comment (CecileRobertMichon) [inline hack/create-dev-cluster.sh:54]: can we remove them from the other script? we shouldn't need to set them in both

Comment (nader-ziada) [reply hack/create-dev-cluster.sh:54]: If we do that, running make create management cluster without the script would not work

Comment (CecileRobertMichon) [reply hack/create-dev-cluster.sh:54]: that's true but currently running make create-cluster doesn't work without the variables set below either

Comment (nader-ziada) [reply hack/create-dev-cluster.sh:54]: ok, moved them from the other script

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**